integration-tests/olminstall/helpers/send_notification.py

```
@@ -0,0 +1,116 @@
+            "WARN: SLACK_WEBHOOK_URL must be https://hooks.slack.com/services/... — skipping Slack",
+            file=sys.stderr,
+        )
+        return 0
+
+    payload = json.dumps({"text": msg, "channel": channel_id}).encode()
+    req = Request(
+        webhook_url,
+        data=payload,
+        headers={"Content-Type": "application/json"},
+        method="POST",
+    )
```

Ruff (0.15.12)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)